web/src/components/tasks/business-formation/review/ReviewSignatures.tsx

```diff
@@ -121,18 +120,18 @@ export const ReviewSignatures = (): ReactElement => {
   };
 
   const signers = (): ReactElement => {
-    if (!isCorp && !isNonprofit && hasNoSigners) {
-      return displayEmptySigners();
-    } else {
+    if (hasSigners || isCorp || isNonprofit) {
       return displaySigners();
+    } else {
+      return displayEmptySigners();
     }
   };
 
-  const incorporators = (): ReactElement => {
-    if ((isCorp || isNonprofit) && hasNoIncorporators) {
-      return displayEmptyIncorporators();
-    } else {
+  const incorporators = (): ReactElement | undefined => {
+    if (hasIncorporators) {
       return displayIncorporators();
+    } else if (isCorp || isNonprofit) {
+      return displayEmptyIncorporators();
```

Comment:
The reason I changed this was follow the same pattern as the signers function. So before the logic made it so `displayEmptyIncorporators` would render if there were no incorporators and if legal type was `isCorp` or `isNonprofit` and `displayIncorporators` would show up as long as there were incorporators. I just flipped and we still display incorporators as long as they are there with the `hasIncorporators` boolean and its the same logic as before where `displayEmptyIncorporators` only shows where the legal structure is either corp or nonprofit and the not having incorporators is implied as the previous `if` would have had to fail to come to the` else if` condition. Am I missing an edge case?

Reply:
@somabadri I'm struggling to think through this tbh (which is why the refactor is so needed) :)

Here's a truth table I'm seeing to help parse it out

OLD VERSION:

|  | Corp / Nonprofit  | LLC / LP / LLP |
| --- | --- | --- |
| Has Incorporators  | displayIncorporators  | displayIncorporators | 
| Has No Incorporators  | displayEmptyIncorporators  | displayIncorporators | 

NEW VERSION:

|  | Corp / Nonprofit  | LLC / LP / LLP |
| --- | --- | -- |
| Has Incorporators  | displayIncorporators  | displayIncorporators | 
| Has No Incorporators  | displayEmptyIncorporators  | ?? display null ?? |

Is this right?

Reply:
To be honest, I don't really know if the "old" logic is correct tho

Reply:
The truth table is correct and since `incorporators` only exist for corporations or non profits we can display nothing in that situation. The only reason the the old logic worked despite us rendering `displayIncorporators` even though there are no incorporators and the user is a LLC/LP/LLP is because the `displayIncorporators` function maps over `state.formationFormData.incorporators` which is empty so it renders nothing.

Reply:
Got it. 

Okay, can I suggest that we make the else case explicit, so that we make it clear the expected behavior for `else`? Basically adding an "else return null" to your if/else-if.

And I do recommend `null` instead of `undefined` (and changing the return type appropriately) just because null is kind of considered the pattern for what to return when NOT returning a react element.